### PR TITLE
[pidcontroller] Remove limits, make Ki dependent from the loop time, add reset command

### PR DESCRIPTION
**Pull request by fwolter**
Also fix naming of thread and shutdown executor.

Signed-off-by: Fabian Wolter <github@fabian-wolter.de>

### CODE REVIEW

File: bundles/org.openhab.automation.pidcontroller/README.md [hunk 1]
```diff
@@ -22,20 +22,21 @@ This module triggers whenever the `input` or the `setpoint` changes or the `loop
...
+| `setpoint`           | Item    | Name of the setpoint Item (e.g. desired room temperature)                                                                                          | Y        |
+| `kp`                 | Decimal | P: [Proportional Gain](#proportional-p-gain-parameter) Parameter                                                                                   | Y        |
+| `ki`                 | Decimal | I: [Integral Gain](#integral-i-gain-parameter) Parameter                                                                                           | Y        |
+| `kd`                 | Decimal | D: [Derivative Gain](#derivative-d-gain-parameter) Parameter                                                                                       | Y        |
+| `kdTimeConstant`     | Decimal | D-T1: [Derivative Gain Time Constant](#derivative-time-constant-d-t1-parameter) in sec.                                                            | Y        |
+| `integralLowerLimit` | Decimal | The I part of the PID controller will be max this value                                                                                            | Y        |
```
Comment (J-N-K) [inline bundles/org.openhab.automation.pidcontroller/README.md:33]: What is the benefit of renaming? I guess that `outputLowerLimit` is more clear

Comment (fwolter) [reply bundles/org.openhab.automation.pidcontroller/README.md:33]: The parameter does not limit the output anymore, but only the integral part.

File: bundles/org.openhab.automation.pidcontroller/src/main/java/org/openhab/automation/pidcontroller/internal/handler/PIDControllerTriggerHandler.java [hunk 2]
```diff
@@ -93,13 +93,17 @@ public PIDControllerTriggerHandler(Trigger module, ItemRegistry itemRegistry, Ev
             throw new IllegalArgumentException("Configured setpoint item not found: " + setpointItemName, e);
         }
 
-        double outputLowerLimit = getDoubleFromConfig(config, CONFIG_OUTPUT_LOWER_LIMIT);
-        double outputUpperLimit = getDoubleFromConfig(config, CONFIG_OUTPUT_UPPER_LIMIT);
+        double outputLowerLimit = getDoubleFromConfig(config, CONFIG_INTEGRAL_LOWER_LIMIT);
```
Comment (J-N-K) [inline bundles/org.openhab.automation.pidcontroller/src/main/java/org/openhab/automation/pidcontroller/internal/handler/PIDControllerTriggerHandler.java:96]: But why is this names outputLowerLimit then?

**Comment by J-N-K** [comment]
Also I think a limit for the output is useful.

File: bundles/org.openhab.automation.pidcontroller/src/main/java/org/openhab/automation/pidcontroller/internal/handler/PIDControllerTriggerHandler.java (hunk 2 above)
Comment (fwolter) [reply bundles/org.openhab.automation.pidcontroller/src/main/java/org/openhab/automation/pidcontroller/internal/handler/PIDControllerTriggerHandler.java]: I overlooked that. Thanks!

**Comment by fwolter** [comment]
The opinions differ. A control expert stated in the [forum](https://community.openhab.org/t/new-automation-pid-controller/111229/26?u=fwolter) that even a limit on the I part is hacked. A limit to the output could be applied by the user via a rule.

**Comment by J-N-K** [comment]
Reading through the thread having a "reset" channel would be better than the limitiation. Why don't we implement that one?

**Comment by fwolter** [comment]
I removed the limits and added a reset command.

**Review by J-N-K — APPROVED** [review]

**Pull request merged**